### ISSUE TITLE
junos_interfaces issue

### ISSUE THREAD

**Issue by gdykeman** [ansible-collections/junipernetworks.junos#36]
##### SUMMARY
junos_interfaces resulting in an error when passing in {{ interfaces }} var that is gathered via network_resources

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
junos_interfaces

##### ANSIBLE VERSION

```
➜  ansible_locker git:(feature/banner) ✗ ansible --version
ansible 2.9.6
  config file = /Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg
  configured module search path = [u'/Users/gdykeman/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Library/Python/2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.16 (default, Oct 16 2019, 00:34:56) [GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.37.14)]
```

##### CONFIGURATION
```
DEFAULT_CALLBACK_WHITELIST(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = [u'time']
DEFAULT_FORKS(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = 50
DEFAULT_HOST_LIST(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = [u'/Users/gdykeman/Documents/Ansible/multivendor_network/multivendor_inventory/dev_hosts.yaml']
DEFAULT_NO_TARGET_SYSLOG(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = False
DEFAULT_PRIVATE_KEY_FILE(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = /Users/gdykeman/.ssh/gdykeman-dev-network.pem
DEFAULT_SCP_IF_SSH(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = True
DEFAULT_STDOUT_CALLBACK(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = yaml
HOST_KEY_CHECKING(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = False
INTERPRETER_PYTHON(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = auto_silent
PERSISTENT_COMMAND_TIMEOUT(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = 200
PERSISTENT_CONNECT_TIMEOUT(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = 200
RETRY_FILES_ENABLED(/Users/gdykeman/Documents/Ansible/ansible_locker/ansible.cfg) = False
```

##### OS / ENVIRONMENT
MacOS

##### STEPS TO REPRODUCE
```
    interfaces:
      - enabled: true
        name: gr-0/0/0
      - enabled: true
        name: lo0
      - enabled: true
        name: fxp0
```
```
  tasks:
    - name: Configure Interfaces with provided data model
      junos_interfaces:
        config:  "{{ interfaces }}"
        state: replaced
```

##### EXPECTED RESULTS
Configuration of interfaces according to data set

##### ACTUAL RESULTS
```
TASK [Configure Interfaces with provided data model] ************************************************************
fatal: [juniper1]: FAILED! => changed=false
  ansible_facts:
    discovered_interpreter_python: /usr/bin/python
  msg: |-
    <?xml version="1.0" encoding="UTF-8"?><rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" xmlns:junos="http://xml.juniper.net/junos/18.4R1/junos" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:ed9ac575-d07e-48a2-94ba-0c4f1809ae43">
    <load-configuration-results>
    <rpc-error>
    <error-severity>warning</error-severity>
    <error-path>[edit interfaces gr-0/0/0]</error-path>
    <error-message>statement not found: description</error-message>
    </rpc-error>
    <rpc-error>
    <error-type>protocol</error-type>
    <error-tag>operation-failed</error-tag>
    <error-severity>error</error-severity>
    <error-message>syntax error</error-message>
    <error-info>
    <bad-element>speed</bad-element>
    </error-info>
    </rpc-error>
    <load-error-count>2</load-error-count>
    </load-configuration-results>
    </rpc-reply>
```

**Comment by danielmellado** [comment]
Hi @gdykeman, thanks for your issue. JunOS doesn't support setting the interface speed over g (Generic routing encapsulation (GRE) tunnel) interfaces, so this the error you're seeing over the return xml. It is *not* an ansible issue but an unsupported usage of the module. If you change gr to a usual ge interface, it'll work as intented.

**Comment by danielmellado** [comment]
Actually I took a deeper look and we were missing some options related to GRE interfaces

**Comment by danielmellado** [comment]
Added a fix in both collection and ansible stable/2.9 branch

**Comment by gdykeman** [comment]
@danielmellado Thank you for looking into that.  This might also be the reason I get a similar error when using gather_facts against the same device.
```
- name: gather_facts
   junos_facts:
     gather_subset: min
     gather_network_resources: all
```
```
    <rpc-error>
    <error-type>protocol</error-type>
    <error-tag>operation-failed</error-tag>
    <error-severity>error</error-severity>
    <error-message>syntax error</error-message>
    <error-info>
    <bad-element>interface</bad-element>
    </error-info>
    </rpc-error>
    <rpc-error>
    <error-type>protocol</error-type>
    <error-tag>operation-failed</error-tag>
    <error-severity>error</error-severity>
    <error-message>syntax error</error-message>
    <error-info>
    <bad-element>interface</bad-element>
    </error-info>
    </rpc-error>
    </rpc-reply>
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1
```
Results in a similar error

**Comment by danielmellado** [comment]
Hi @gdykeman, did you get to see this? https://github.com/ansible-collections/junipernetworks.junos/issues/17. I would need to see some more detailed data there, as I can't reproduce your issue nor locally or in CI

**Comment by danielmellado** [comment]
Would you mind sharing more of your trace?

**Comment by gdykeman** [comment]
@danielmellado my apologies. Here's the full error:
```
TASK [gather_facts] ****************************************************************************************************************************
task path: /Users/gdykeman/Documents/Ansible/ansible_locker/test.yaml:5
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: </rpc-reply>
fatal: [juniper1]: FAILED! => changed=false
  ansible_facts:
    discovered_interpreter_python: /usr/bin/python
  module_stderr: |-
    Traceback (most recent call last):
      File "/Users/gdykeman/.ansible/tmp/ansible-local-14640Xm_gl6/ansible-tmp-1589196058.6-3694611333333/AnsiballZ_junos_facts.py", line 102, in <module>
        _ansiballz_main()
      File "/Users/gdykeman/.ansible/tmp/ansible-local-14640Xm_gl6/ansible-tmp-1589196058.6-3694611333333/AnsiballZ_junos_facts.py", line 94, in _ansiballz_main
        invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
      File "/Users/gdykeman/.ansible/tmp/ansible-local-14640Xm_gl6/ansible-tmp-1589196058.6-3694611333333/AnsiballZ_junos_facts.py", line 40, in invoke_module
        runpy.run_module(mod_name='ansible.modules.network.junos.junos_facts', init_globals=None, run_name='__main__', alter_sys=True)
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 188, in run_module
        fname, loader, pkg_name)
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 82, in _run_module_code
        mod_name, mod_fname, mod_loader, pkg_name)
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
        exec code in run_globals
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/modules/network/junos/junos_facts.py", line 135, in <module>
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/modules/network/junos/junos_facts.py", line 126, in main
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/module_utils/network/junos/facts/facts.py", line 62, in get_facts
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/module_utils/network/common/facts/facts.py", line 105, in get_network_resources_facts
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/module_utils/network/junos/facts/lldp_interfaces/lldp_interfaces.py", line 69, in populate_facts
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/module_utils/network/junos/utils/utils.py", line 28, in get_resource_config
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/module_utils/network/common/netconf.py", line 76, in __rpc__
      File "/var/folders/51/9ngxm1zx0x731370688ykt7c0000gn/T/ansible_junos_facts_payload_h55qyL/ansible_junos_facts_payload.zip/ansible/module_utils/network/common/netconf.py", line 105, in parse_rpc_error
    ansible.module_utils.connection.ConnectionError: <?xml version="1.0" encoding="UTF-8"?><rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" xmlns:junos="http://xml.juniper.net/junos/18.4R1/junos" xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:8afc1f54-d732-45bd-90aa-77151b0b89fe">
    <rpc-error>
    <error-type>protocol</error-type>
    <error-tag>operation-failed</error-tag>
    <error-severity>error</error-severity>
    <error-message>syntax error</error-message>
    <error-info>
    <bad-element>interface</bad-element>
    </error-info>
    </rpc-error>
    <rpc-error>
    <error-type>protocol</error-type>
    <error-tag>operation-failed</error-tag>
    <error-severity>error</error-severity>
    <error-message>syntax error</error-message>
    <error-info>
    <bad-element>interface</bad-element>
    </error-info>
    </rpc-error>
    </rpc-reply>
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1

PLAY RECAP *************************************************************************************************************************************
juniper1                   : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

**Comment by danielmellado** [comment]
Hey @gdykeman, so, after checking a bit, it does seem that ansible galaxy installs a different version of the collection than the one on the git repo here. What you're seeing, and I could reproduce, is an issue which got fixed by #20 

It doesn't seem, though, that galaxy offers such fix, so I'll check. In the meanwhile, you can just grab the latest version of the collection from the git repo directly and put that on your collections path.
